meilisearch/tests/search/multi.rs

```diff
@@ -376,3 +1097,3227 @@ async fn search_multiple_query_errors() {
+async fn federation_sort_same_indexes_same_criterion_opposite_direction() {
+    let server = Server::new().await;
+
+    let index = server.index("nested");
+
+    let documents = NESTED_DOCUMENTS.clone();
+    let (value, _) = index.add_documents(documents, None).await;
+    index.wait_task(value.uid()).await;
+
+    let (value, _) = index
+        .update_settings(json!({
+          "sortableAttributes": ["mother"],
+          "rankingRules": [
+            "sort",
+            "words",
+            "typo",
+            "proximity",
+            "attribute",
+            "exactness"
+          ]
+        }))
+        .await;
+    index.wait_task(value.uid()).await;
+
+    // two identical placeholder search should have all results from first query
+    let (response, code) = server
+        .multi_search(json!({"federation": {}, "queries": [
+          {"indexUid" : "nested", "q": "", "sort": ["mother:asc"], "showRankingScore": true },
+          {"indexUid" : "nested", "q": "", "sort": ["mother:desc"], "showRankingScore": true },
+        ]}))
+        .await;
+    snapshot!(code, @"400 Bad Request");
+    insta::assert_json_snapshot!(response, { ".processingTimeMs" => "[time]" }, @r###"
+    {
+      "message": "Inside `.queries[1]`: The results of queries #0 and #1 are incompatible: \n  1. `queries[0].sort[0]`, `nested.rankingRules[0]`: ascending sort rule(s) on field `mother`\n  2. `queries[1].sort[0]`, `nested.rankingRules[0]`: descending sort rule(s) on field `mother`\n  - cannot compare two sort rules in opposite directions\n  - note: The ranking rules of query #0 were modified during canonicalization:\n    1. Removed relevancy rule `words` at position #1 in ranking rules because the query is a placeholder search (`q`: \"\")\n    2. Removed relevancy rule `typo` at position #2 in ranking rules because the query is a placeholder search (`q`: \"\")\n    3. Removed relevancy rule `proximity` at position #3 in ranking rules because the query is a placeholder search (`q`: \"\")\n    4. Removed relevancy rule `attribute` at position #4 in ranking rules because the query is a placeholder search (`q`: \"\")\n    5. Removed relevancy rule `exactness` at position #5 in ranking rules because the query is a placeholder search (`q`: \"\")\n  - note: The ranking rules of query #1 were modified during canonicalization:\n    1. Removed relevancy rule `words` at position #1 in ranking rules because the query is a placeholder search (`q`: \"\")\n    2. Removed relevancy rule `typo` at position #2 in ranking rules because the query is a placeholder search (`q`: \"\")\n    3. Removed relevancy rule `proximity` at position #3 in ranking rules because the query is a placeholder search (`q`: \"\")\n    4. Removed relevancy rule `attribute` at position #4 in ranking rules because the query is a placeholder search (`q`: \"\")\n    5. Removed relevancy rule `exactness` at position #5 in ranking rules because the query is a placeholder search (`q`: \"\")\n",
```

Comment:
Well, it's a bit long, but it's clear. 
I think the `note` parts are a bit too much, but on the other hand, there is nowhere else to put it.
Maybe a log::info or debug?

Reply:
Hello, thanks for your comment.

I think the note parts are very important to understand what is happening in "nonstandard" cases.
Putting them in the logs is a bad location, because users don't look at logs. Currently, Cloud users cannot even look at logs. I think it is best if users can troubleshoot their issues themselves, and so we need to give them the tools to do so.

Now, you've got a point that the output is verbose. The thing could be made less verbose by displaying only notes that are responsible for a rule mismatch 🤔 I'm not 100% sure how to do that, however

Reply:
Yes, I agree it can be useful, let's keep it and see